### ISSUE TITLE
Changing version per project feature does not work

### ISSUE THREAD

**Issue by 4513ECHO** [aquaproj/aqua#1106]
## aqua version

Please use the latest version.

```console
$ aqua -v
aqua version 1.19.1 (3cf7001c3ab8a0333765de44c14205746eb806f4)
```

## Environment

* OS (Windows, Linux, macOS, etc):

Linux (Ubuntu 21.04)

* CPU Architecture (amd64, arm64, etc):

arm64

## Overview

aqua does not detect and change config per project automatically.

## How to reproduce

Please run my [gist](https://gist.github.com/7659eea428a15551b261694e54631865).

```console
$ curl -fsSL https://gist.github.com/raw/7659eea428a15551b261694e54631865 | sh
```

## Debug Output

Please set the environment variable `AQUA_LOG_LEVEL=debug` and run aqua and tell us the debug output.

```console
$ curl -fsSL https://gist.github.com/raw/7659eea428a15551b261694e54631865 | sh
+ export AQUA_LOG_LEVEL=debug
+ aqua --version
aqua version 1.19.1 (3cf7001c3ab8a0333765de44c14205746eb806f4)
+ mkdir foo bar
+ cat
+ tee foo/aqua.yaml
# foo/aqua.yaml
registries:
- type: standard
  ref: v3.10.0 # renovate: depName=aquaproj/aqua-registry

packages:
- name: cli/cli@v2.1.0
+ + tee bar/aqua.yaml
cat
# bar/aqua.yaml
registries:
- type: standard
  ref: v3.10.0 # renovate: depName=aquaproj/aqua-registry

packages:
- name: cli/cli@v2.0.0
+ cd foo
+ pwd
/home/hibiki/Develops/github.com/aquaproj/aqua/foo
+ gh version
DEBU[0000] install the package                           aqua_version=1.19.1 env=linux/arm64 exe_path=/home/hibiki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.14.4/gh_2.14.4_linux_arm64.tar.gz/gh_2.14.4_linux_arm64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.14.4 program=aqua registry=standard
DEBU[0000] check if the package is already installed     aqua_version=1.19.1 env=linux/arm64 exe_path=/home/hibiki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.14.4/gh_2.14.4_linux_arm64.tar.gz/gh_2.14.4_linux_arm64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.14.4 program=aqua registry=standard
DEBU[0000] check the permission                          aqua_version=1.19.1 env=linux/arm64 exe_path=/home/hibiki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.14.4/gh_2.14.4_linux_arm64.tar.gz/gh_2.14.4_linux_arm64/bin/gh file_name=gh package=cli/cli package_name=cli/cli package_version=v2.14.4 program=aqua registry=standard
DEBU[0000] check if exec file exists                     aqua_version=1.19.1 env=linux/arm64 exe_path=/home/hibiki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.14.4/gh_2.14.4_linux_arm64.tar.gz/gh_2.14.4_linux_arm64/bin/gh package=cli/cli program=aqua
DEBU[0000] execute the command                           aqua_version=1.19.1 env=linux/arm64 exe_path=/home/hibiki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.14.4/gh_2.14.4_linux_arm64.tar.gz/gh_2.14.4_linux_arm64/bin/gh program=aqua
gh version 2.14.4 (2022-08-10)
https://github.com/cli/cli/releases/tag/v2.14.4
+ cd ../bar
+ pwd
/home/hibiki/Develops/github.com/aquaproj/aqua/bar
+ gh version
DEBU[0000] install the package                           aqua_version=1.19.1 env=linux/arm64 exe_path=/home/hibiki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.14.4/gh_2.14.4_linux_arm64.tar.gz/gh_2.14.4_linux_arm64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.14.4 program=aqua registry=standard
DEBU[0000] check if the package is already installed     aqua_version=1.19.1 env=linux/arm64 exe_path=/home/hibiki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.14.4/gh_2.14.4_linux_arm64.tar.gz/gh_2.14.4_linux_arm64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.14.4 program=aqua registry=standard
DEBU[0000] check the permission                          aqua_version=1.19.1 env=linux/arm64 exe_path=/home/hibiki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.14.4/gh_2.14.4_linux_arm64.tar.gz/gh_2.14.4_linux_arm64/bin/gh file_name=gh package=cli/cli package_name=cli/cli package_version=v2.14.4 program=aqua registry=standard
DEBU[0000] check if exec file exists                     aqua_version=1.19.1 env=linux/arm64 exe_path=/home/hibiki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.14.4/gh_2.14.4_linux_arm64.tar.gz/gh_2.14.4_linux_arm64/bin/gh package=cli/cli program=aqua
DEBU[0000] execute the command                           aqua_version=1.19.1 env=linux/arm64 exe_path=/home/hibiki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.14.4/gh_2.14.4_linux_arm64.tar.gz/gh_2.14.4_linux_arm64/bin/gh program=aqua
gh version 2.14.4 (2022-08-10)
https://github.com/cli/cli/releases/tag/v2.14.4
```

## Expected Behaviour

In `foo/` `gh version` should show `2.1.0`, and `2.0.0` in `bar/`.

## Actual Behaviour

Both of directories show `2.14.4`, which currect version of global config.

## Important Factoids

If `$AQUA_CONFIG` or `--config` parameter is set, aqua will change version per project correctly.

## References

* https://aquaproj.github.io/docs/tutorial-basics/change-version-per-project/

**Comment by suzuki-shunsuke** [comment]
Thank you for your report.

I can't reproduce the issue in my laptop (darwin/arm64). I'll confirm linux/arm64 later.

```console
$ curl -fsSL https://gist.github.com/raw/7659eea428a15551b261694e54631865 | sh
+ export AQUA_LOG_LEVEL=debug
+ AQUA_LOG_LEVEL=debug
+ aqua --version
aqua version 1.19.1 (3cf7001c3ab8a0333765de44c14205746eb806f4)
+ mkdir foo bar
+ cat
+ tee foo/aqua.yaml
# foo/aqua.yaml
registries:
- type: standard
  ref: v3.10.0 # renovate: depName=aquaproj/aqua-registry

packages:
- name: cli/cli@v2.1.0
+ cat
+ tee bar/aqua.yaml
# bar/aqua.yaml
registries:
- type: standard
  ref: v3.10.0 # renovate: depName=aquaproj/aqua-registry

packages:
- name: cli/cli@v2.0.0
+ cd foo
+ pwd
/Users/shunsukesuzuki/Documents/test/aqua/issue/foo
+ gh version
DEBU[0000] install the package                           aqua_version=1.19.1 env=darwin/arm64 exe_path=/Users/shunsukesuzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.1.0/gh_2.1.0_macOS_amd64.tar.gz/gh_2.1.0_macOS_amd64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.1.0 program=aqua registry=standard
DEBU[0000] check if the package is already installed     aqua_version=1.19.1 env=darwin/arm64 exe_path=/Users/shunsukesuzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.1.0/gh_2.1.0_macOS_amd64.tar.gz/gh_2.1.0_macOS_amd64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.1.0 program=aqua registry=standard
INFO[0000] download and unarchive the package            aqua_version=1.19.1 env=darwin/arm64 exe_path=/Users/shunsukesuzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.1.0/gh_2.1.0_macOS_amd64.tar.gz/gh_2.1.0_macOS_amd64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.1.0 program=aqua registry=standard
DEBU[0001] check the permission                          aqua_version=1.19.1 env=darwin/arm64 exe_path=/Users/shunsukesuzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.1.0/gh_2.1.0_macOS_amd64.tar.gz/gh_2.1.0_macOS_amd64/bin/gh file_name=gh package=cli/cli package_name=cli/cli package_version=v2.1.0 program=aqua registry=standard
DEBU[0001] check if exec file exists                     aqua_version=1.19.1 env=darwin/arm64 exe_path=/Users/shunsukesuzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.1.0/gh_2.1.0_macOS_amd64.tar.gz/gh_2.1.0_macOS_amd64/bin/gh package=cli/cli program=aqua
DEBU[0001] execute the command                           aqua_version=1.19.1 env=darwin/arm64 exe_path=/Users/shunsukesuzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.1.0/gh_2.1.0_macOS_amd64.tar.gz/gh_2.1.0_macOS_amd64/bin/gh program=aqua
gh version 2.1.0 (2021-10-14)
https://github.com/cli/cli/releases/tag/v2.1.0


A new release of gh is available: 2.1.0 → v2.14.4
https://github.com/cli/cli/releases/tag/v2.14.4

+ cd ../bar
+ pwd
/Users/shunsukesuzuki/Documents/test/aqua/issue/bar
+ gh version
DEBU[0000] install the package                           aqua_version=1.19.1 env=darwin/arm64 exe_path=/Users/shunsukesuzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.0.0/gh_2.0.0_macOS_amd64.tar.gz/gh_2.0.0_macOS_amd64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.0.0 program=aqua registry=standard
DEBU[0000] check if the package is already installed     aqua_version=1.19.1 env=darwin/arm64 exe_path=/Users/shunsukesuzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.0.0/gh_2.0.0_macOS_amd64.tar.gz/gh_2.0.0_macOS_amd64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.0.0 program=aqua registry=standard
INFO[0000] download and unarchive the package            aqua_version=1.19.1 env=darwin/arm64 exe_path=/Users/shunsukesuzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.0.0/gh_2.0.0_macOS_amd64.tar.gz/gh_2.0.0_macOS_amd64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.0.0 program=aqua registry=standard
DEBU[0001] check the permission                          aqua_version=1.19.1 env=darwin/arm64 exe_path=/Users/shunsukesuzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.0.0/gh_2.0.0_macOS_amd64.tar.gz/gh_2.0.0_macOS_amd64/bin/gh file_name=gh package=cli/cli package_name=cli/cli package_version=v2.0.0 program=aqua registry=standard
DEBU[0001] check if exec file exists                     aqua_version=1.19.1 env=darwin/arm64 exe_path=/Users/shunsukesuzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.0.0/gh_2.0.0_macOS_amd64.tar.gz/gh_2.0.0_macOS_amd64/bin/gh package=cli/cli program=aqua
DEBU[0001] execute the command                           aqua_version=1.19.1 env=darwin/arm64 exe_path=/Users/shunsukesuzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.0.0/gh_2.0.0_macOS_amd64.tar.gz/gh_2.0.0_macOS_amd64/bin/gh program=aqua
gh version 2.0.0 (2021-08-24)
https://github.com/cli/cli/releases/tag/v2.0.0
```

```console
$ cd foo 

$ cat aqua.yaml 
# foo/aqua.yaml
registries:
- type: standard
  ref: v3.10.0 # renovate: depName=aquaproj/aqua-registry

packages:
- name: cli/cli@v2.1.0

$ gh version
gh version 2.1.0 (2021-10-14)
https://github.com/cli/cli/releases/tag/v2.1.0

$ cd ../bar 

$ cat aqua.yaml 
# bar/aqua.yaml
registries:
- type: standard
  ref: v3.10.0 # renovate: depName=aquaproj/aqua-registry

packages:
- name: cli/cli@v2.0.0

$ gh version
gh version 2.0.0 (2021-08-24)
https://github.com/cli/cli/releases/tag/v2.0.0
```

**Comment by suzuki-shunsuke** [comment]
I can't reproduce the issue in linux/arm64 too.

```console
shunsukesuzuki@lima-default:~/test/1106$ curl -fsSL https://gist.github.com/raw/7659eea428a15551b261694e54631865 | sh
+ export AQUA_LOG_LEVEL=debug
+ aqua --version
aqua version 1.19.1 (3cf7001c3ab8a0333765de44c14205746eb806f4)
+ mkdir foo bar
+ + cat
tee foo/aqua.yaml
# foo/aqua.yaml
registries:
- type: standard
  ref: v3.10.0 # renovate: depName=aquaproj/aqua-registry

packages:
- name: cli/cli@v2.1.0
+ + cat
tee bar/aqua.yaml
# bar/aqua.yaml
registries:
- type: standard
  ref: v3.10.0 # renovate: depName=aquaproj/aqua-registry

packages:
- name: cli/cli@v2.0.0
+ cd foo
+ pwd
/home/shunsukesuzuki.linux/test/1106/foo
+ gh version
DEBU[0001] install the package                           aqua_version=1.19.1 env=linux/arm64 exe_path=/home/shunsukesuzuki.linux/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.1.0/gh_2.1.0_linux_arm64.tar.gz/gh_2.1.0_linux_arm64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.1.0 program=aqua registry=standard
DEBU[0001] check if the package is already installed     aqua_version=1.19.1 env=linux/arm64 exe_path=/home/shunsukesuzuki.linux/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.1.0/gh_2.1.0_linux_arm64.tar.gz/gh_2.1.0_linux_arm64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.1.0 program=aqua registry=standard
INFO[0001] download and unarchive the package            aqua_version=1.19.1 env=linux/arm64 exe_path=/home/shunsukesuzuki.linux/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.1.0/gh_2.1.0_linux_arm64.tar.gz/gh_2.1.0_linux_arm64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.1.0 program=aqua registry=standard
DEBU[0002] check the permission                          aqua_version=1.19.1 env=linux/arm64 exe_path=/home/shunsukesuzuki.linux/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.1.0/gh_2.1.0_linux_arm64.tar.gz/gh_2.1.0_linux_arm64/bin/gh file_name=gh package=cli/cli package_name=cli/cli package_version=v2.1.0 program=aqua registry=standard
DEBU[0002] check if exec file exists                     aqua_version=1.19.1 env=linux/arm64 exe_path=/home/shunsukesuzuki.linux/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.1.0/gh_2.1.0_linux_arm64.tar.gz/gh_2.1.0_linux_arm64/bin/gh package=cli/cli program=aqua
DEBU[0002] execute the command                           aqua_version=1.19.1 env=linux/arm64 exe_path=/home/shunsukesuzuki.linux/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.1.0/gh_2.1.0_linux_arm64.tar.gz/gh_2.1.0_linux_arm64/bin/gh program=aqua
gh version 2.1.0 (2021-10-14)
https://github.com/cli/cli/releases/tag/v2.1.0


A new release of gh is available: 2.1.0 → v2.14.4
https://github.com/cli/cli/releases/tag/v2.14.4

+ cd ../bar
+ pwd
/home/shunsukesuzuki.linux/test/1106/bar
+ gh version
DEBU[0000] install the package                           aqua_version=1.19.1 env=linux/arm64 exe_path=/home/shunsukesuzuki.linux/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.0.0/gh_2.0.0_linux_arm64.tar.gz/gh_2.0.0_linux_arm64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.0.0 program=aqua registry=standard
DEBU[0000] check if the package is already installed     aqua_version=1.19.1 env=linux/arm64 exe_path=/home/shunsukesuzuki.linux/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.0.0/gh_2.0.0_linux_arm64.tar.gz/gh_2.0.0_linux_arm64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.0.0 program=aqua registry=standard
INFO[0000] download and unarchive the package            aqua_version=1.19.1 env=linux/arm64 exe_path=/home/shunsukesuzuki.linux/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.0.0/gh_2.0.0_linux_arm64.tar.gz/gh_2.0.0_linux_arm64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.0.0 program=aqua registry=standard
DEBU[0001] check the permission                          aqua_version=1.19.1 env=linux/arm64 exe_path=/home/shunsukesuzuki.linux/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.0.0/gh_2.0.0_linux_arm64.tar.gz/gh_2.0.0_linux_arm64/bin/gh file_name=gh package=cli/cli package_name=cli/cli package_version=v2.0.0 program=aqua registry=standard
DEBU[0001] check if exec file exists                     aqua_version=1.19.1 env=linux/arm64 exe_path=/home/shunsukesuzuki.linux/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.0.0/gh_2.0.0_linux_arm64.tar.gz/gh_2.0.0_linux_arm64/bin/gh package=cli/cli program=aqua
DEBU[0001] execute the command                           aqua_version=1.19.1 env=linux/arm64 exe_path=/home/shunsukesuzuki.linux/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.0.0/gh_2.0.0_linux_arm64.tar.gz/gh_2.0.0_linux_arm64/bin/gh program=aqua
gh version 2.0.0 (2021-08-24)
https://github.com/cli/cli/releases/tag/v2.0.0
```

```console
shunsukesuzuki@lima-default:~/test/1106$ echo $AQUA_GLOBAL_CONFIG 
/Users/shunsukesuzuki/repos/src/github.com/aquaproj/aqua-registry/aqua-all.yaml
shunsukesuzuki@lima-default:~/test/1106$ aqua -v
aqua version 1.19.1 (3cf7001c3ab8a0333765de44c14205746eb806f4)
shunsukesuzuki@lima-default:~/test/1106$ cd foo/
shunsukesuzuki@lima-default:~/test/1106/foo$ cat aqua.yaml 
# foo/aqua.yaml
registries:
- type: standard
  ref: v3.10.0 # renovate: depName=aquaproj/aqua-registry

packages:
- name: cli/cli@v2.1.0
shunsukesuzuki@lima-default:~/test/1106/foo$ gh version
gh version 2.1.0 (2021-10-14)
https://github.com/cli/cli/releases/tag/v2.1.0
shunsukesuzuki@lima-default:~/test/1106/foo$ cd ../bar/
shunsukesuzuki@lima-default:~/test/1106/bar$ cat aqua.yaml 
# bar/aqua.yaml
registries:
- type: standard
  ref: v3.10.0 # renovate: depName=aquaproj/aqua-registry

packages:
- name: cli/cli@v2.0.0
shunsukesuzuki@lima-default:~/test/1106/bar$ gh version
gh version 2.0.0 (2021-08-24)
https://github.com/cli/cli/releases/tag/v2.0.0
shunsukesuzuki@lima-default:~/test/1106/bar$ cd ..
shunsukesuzuki@lima-default:~/test/1106$ gh version
INFO[0003] download and unarchive the package            aqua_version=1.19.1 env=linux/arm64 exe_path=/home/shunsukesuzuki.linux/.local/share/aquaproj-aqua/pkgs/github_release/github.com/cli/cli/v2.14.3/gh_2.14.3_linux_arm64.tar.gz/gh_2.14.3_linux_arm64/bin/gh package=cli/cli package_name=cli/cli package_version=v2.14.3 program=aqua registry=standard
gh version 2.14.3 (2022-07-26)
https://github.com/cli/cli/releases/tag/v2.14.3

**Comment by 4513ECHO** [comment]
I've updated the gist to echo `$AQUA_GLOBLAL_CONFIG` and `$AQUA_CONFIG`, and I noticed `$AQUA_CONFIG` was set without noticing.
After I run `unset AQUA_CONFIG`, I got expected behavior. Sorry.
